python/modules/map_plate.py

```diff
@@ -7,30 +7,65 @@
 logger = logging.getLogger("map plate")
 logger.info("Executing map_plate module.")
 
-def _preparing_output_csv(input_data, output_path, mp_data, names, mp_deltimer, csv_deltimer):
-    output_data = []
-    num = 0
-    position = 0 
-    for line in input_data:
-        if num == 0: 
-            new_line = line
-            for i in range(len(line)):
-                if line[i] == "well.name":
-                    position = i
-            for j in names:
-                new_line.append(j)
-            output_data.append(new_line)
-            num = 1
+def parse_mp(input_path, delimiter):
+    mp_dict = {}
+    dim =_parse_base_params(input_path, delimiter, mp_dict)
+    _parse_params(input_path, delimiter, dim, mp_dict)
+    return mp_dict
+
+def _parse_params(input_path, delimiter, dimensions, mp_dict):
+    param_paths =_get_param_paths(input_path)
+    param_paths = sorted(param_paths, key = str)
+    logger.debug("Map_plate params input paths: %s.", param_paths)
+    for mp_file in param_paths:
+        correctness = CSV_M.verify_input_file(mp_file, delimiter, dimensions)
+        if correctness == False:
+            logger.error("Fatal error occured while trying to read mapplate.")
```

Comment:
I can't agree with you. Funcion verify_input_file also provides logs messages and we end up with pretty informative and clear message.